### ISSUE TITLE
CocoaPods version incompatibility results in stack trace

### ISSUE THREAD

**Issue by cbracken** [flutter/flutter#8685]
## Steps to Reproduce

If an incompatible version of CocoaPods is installed, the `pod install` step in iOS build/run errors out with a stack trace rather than a clean exit (or warning that `pod install` wasn't successful).

From @collinjackson:

```
Analyzing dependencies
[!] The version of CocoaPods used to generate the lockfile (1.0.1) is higher than the version of the current executable (0.35.0). Incompatibility issues may arise.
[!] The `master` repo requires CocoaPods 1.0.0 -  (currently using 0.35.0)
Exception from flutter run: Exit code 1 from: pod install
Update CocoaPods, or checkout the appropriate tag in the repo.
package:flutter_tools/src/base/process.dart 294                   _runWithLoggingSync
package:flutter_tools/src/base/process.dart 216                   runCheckedSync
package:flutter_tools/src/ios/mac.dart 332                        _installCocoaPods
package:flutter_tools/src/ios/mac.dart 129                        buildXcodeProject
package:flutter_tools/src/ios/simulators.dart 518                 IOSSimulator._buildAndInstallApplicationBundle
package:flutter_tools/src/ios/simulators.dart 512                 IOSSimulator._setupUpdatedApplicationBundle
package:flutter_tools/src/ios/simulators.dart 433                 IOSSimulator.startApp
package:flutter_tools/src/run_hot.dart 157                        HotRunner._run
package:flutter_tools/src/run_hot.dart 78                         HotRunner.run.<fn>
package:stack_trace/src/chain.dart 93                             Chain.capture.<fn>
dart:async/zone.dart 1150                                         _rootRun
dart:async/zone.dart 1026                                         _CustomZone.run
dart:async/zone.dart 1503                                         runZoned
package:stack_trace/src/chain.dart 91                             Chain.capture
package:flutter_tools/src/run_hot.dart 77                         HotRunner.run
package:flutter_tools/src/commands/daemon.dart 406                AppDomain.startApp.<fn>
package:flutter_tools/src/base/context.dart 76                    AppContext._run
package:flutter_tools/src/base/context.dart 66                    AppContext.runInZone.<fn>
dart:async/zone.dart 1150                                         _rootRun
dart:async/zone.dart 1026                                         _CustomZone.run
dart:async/zone.dart 1503                                         runZoned
package:flutter_tools/src/base/context.dart 65                    AppContext.runInZone
package:flutter_tools/src/commands/daemon.dart 732                AppInstance._runInZone
package:flutter_tools/src/commands/daemon.dart 404                AppDomain.startApp
package:flutter_tools/src/commands/run.dart 216                   RunCommand.runCommand
package:flutter_tools/src/runner/flutter_command.dart 148         FlutterCommand.verifyThenRunCommand
package:flutter_tools/src/commands/run.dart 199                   RunCommand.verifyThenRunCommand
package:flutter_tools/src/runner/flutter_command.dart 119         FlutterCommand.run
package:args/command_runner.dart 194                              CommandRunner.runCommand
package:flutter_tools/src/runner/flutter_command_runner.dart 261  FlutterCommandRunner.runCommand
package:args/command_runner.dart 109                              CommandRunner.run.<fn>
dart:async/future.dart 208                                        Future.Future.sync
package:args/command_runner.dart 109                              CommandRunner.run
package:flutter_tools/src/runner/flutter_command_runner.dart 157  FlutterCommandRunner.run
package:flutter_tools/executable.dart 145                         run.<fn>.<fn>
package:stack_trace/src/chain.dart 93                             Chain.capture.<fn>
dart:async/zone.dart 1150                                         _rootRun
dart:async/zone.dart 1026                                         _CustomZone.run
dart:async/zone.dart 1503                                         runZoned
package:stack_trace/src/chain.dart 91                             Chain.capture
package:flutter_tools/executable.dart 144                         run.<fn>
package:flutter_tools/src/base/context.dart 76                    AppContext._run
package:flutter_tools/src/base/context.dart 66                    AppContext.runInZone.<fn>
dart:async/zone.dart 1150                                         _rootRun
dart:async/zone.dart 1026                                         _CustomZone.run
dart:async/zone.dart 1503                                         runZoned
package:flutter_tools/src/base/context.dart 65                    AppContext.runInZone
package:flutter_tools/executable.dart 114                         run
package:flutter_tools/executable.dart 68                          main
../../flutter/packages/flutter_tools/bin/flutter_tools.dart 8     main
dart:isolate-patch/isolate_patch.dart 259                         _startIsolate.<fn>
dart:isolate-patch/isolate_patch.dart 148                         _RawReceivePortImpl._handleMessage
===== asynchronous gap ===========================
dart:async/future_impl.dart 22                                    _Completer.completeError
package:flutter_tools/src/ios/mac.dart 206                        buildXcodeProject
===== asynchronous gap ===========================
dart:async/zone.dart 1073                                         _CustomZone.registerUnaryCallback
dart:async-patch/async_patch.dart 30                              _asyncThenWrapperHelper
package:flutter_tools/src/ios/mac.dart                            buildXcodeProject
package:flutter_tools/src/ios/simulators.dart 518                 IOSSimulator._buildAndInstallApplicationBundle
===== asynchronous gap ===========================
dart:async/zone.dart 1061                                         _CustomZone.registerCallback
dart:async/zone.dart 949                                          _CustomZone.bindCallback
dart:async/schedule_microtask.dart 148                            scheduleMicrotask
dart:async/future.dart 182                                        Future.Future.microtask
package:flutter_tools/src/ios/simulators.dart                     IOSSimulator._buildAndInstallApplicationBundle
package:flutter_tools/src/ios/simulators.dart 512                 IOSSimulator._setupUpdatedApplicationBundle
===== asynchronous gap ===========================
dart:async/zone.dart 1073                                         _CustomZone.registerUnaryCallback
dart:async-patch/async_patch.dart 30                              _asyncThenWrapperHelper
package:flutter_tools/src/ios/simulators.dart                     IOSSimulator._setupUpdatedApplicationBundle
package:flutter_tools/src/ios/simulators.dart 433                 IOSSimulator.startApp
===== asynchronous gap ===========================
dart:async/zone.dart 1061                                         _CustomZone.registerCallback
dart:async/zone.dart 949                                          _CustomZone.bindCallback
dart:async/schedule_microtask.dart 148                            scheduleMicrotask
dart:async/future.dart 182                                        Future.Future.microtask
package:flutter_tools/src/ios/simulators.dart                     IOSSimulator.startApp
package:flutter_tools/src/run_hot.dart 157                        HotRunner._run
===== asynchronous gap ===========================
dart:async/zone.dart 1073                                         _CustomZone.registerUnaryCallback
dart:async-patch/async_patch.dart 30                              _asyncThenWrapperHelper
package:flutter_tools/src/run_hot.dart                            HotRunner._run
package:flutter_tools/src/run_hot.dart 78                         HotRunner.run.<fn>
package:stack_trace/src/chain.dart 93                             Chain.capture.<fn>
dart:async/zone.dart 1150                                         _rootRun
dart:async/zone.dart 1026                                         _CustomZone.run
dart:async/zone.dart 1503                                         runZoned
package:stack_trace/src/chain.dart 91                             Chain.capture
package:flutter_tools/src/run_hot.dart 77                         HotRunner.run
package:flutter_tools/src/commands/daemon.dart 406                AppDomain.startApp.<fn>
```

/cc @jakobr-google

**Comment by cbracken** [comment]
A related issue is that the developer may not want to update their version of CocoaPods for various reasons; e.g. needs to work wither other codebases with older configs they are not ready to upgrade.

**Comment by jakobr-google** [comment]
Running `pod install` doesn't update any pods, it just installs the already selected versions (and selects versions for any pods that hadn't been locked previously). It should be a safe thing to do, and it's necessary to get a smooth experience for native plugins. In other words, *we* will be making use of CocoaPods.

That said, we should fix this issue and give a proper error without the stacktrace. The 0.35.0 version of CocoaPods is about 2 1/2 years old, so you'll probably have to upgrade it, though.

**Comment by jakobr-google** [comment]
Or were you talking about not wanting to upgrade CocoaPods itself? 1.0 was released almost a year ago, and IIRC pre-1.0 releases won't even work with the master repo anymore.

In any case, if they already have a pre-1.0 CocoaPods installed, and it currently works for their app (ie. their Podfile.lock is also created by that version), then things will still continue to work even if we also run pod install automatically.

You're getting the error message in this case because our new project template contains a Podfile.lock created by CocoaPods 1.0.1, and CocoaPods 0.35 doesn't like that. So you'll either need to upgrade CocoaPods, manually remove the Podfile.lock and make sure `pod install` then works (I believe it should, since our new project template has a pretty empty Podfile), or remove the pod integration for the project (if you're not using it).
